lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module

```diff
@@ -60,6 +61,205 @@ function dosomething_rogue_get_reportback_items($inputs)
+
+        try {
+          $campaign_run = dosomething_helpers_get_current_campaign_run_for_user($signup['campaign_id'], $drupal_user); // @TODO - get errors when calling this sometimes it seems (500 - no campaign data found??????)
+        } catch(Exception $e) {
+          // the regular Phoenix endpoint doesn't seem to handle this so should we just leave it? unclear
+          $campaign_run = null;
+        }
+
+        $campaign_run_nid = $campaign_run->nid;
+
+        $current = ($signups['campaign_run_id'] == $campaign_run_nid) ? true : false;
+
+        if (isset($campaign_run->field_run_date['en'][0])) {
+          $current_run_start = $campaign_run->field_run_date['en'][0]['value'];
+          $current_run_end = $campaign_run->field_run_date['en'][0]['value2'];
+        }
+        else {
+          $current_run_start = NULL;
+          $current_run_end = NULL;
+        }
+
+        $campaign = node_load($signup['campaign_id']);
+
+        $campaign_data = (new SignupTransformer)->transformCampaign((object) $campaign);
+
+        $user_data = (new SignupTransformer)->transformUser((object) $northstar_user);
+
+      // Format the data for each signup
+        $signup_data = [
+          // SIGNUP
+          'id' => $signup['signup_id'],
+          'created_at' => strtotime($signup['created_at']),
+
+          // CAMPAIGN RUN
+          'campaign_run' => [
+            'id' => $campaign_run_nid,
+            'current' => $current,
+            'timing' => [
+              'start' => $current_run_start,
+              'end' => $current_run_end,
+            ],
+          ],
+          'uri' => null,
+
+          // CAMPAIGN - no changes
+          'campaign' => $campaign_data,
+
+          // USER - no changes
+          'user' => $user_data,
+        ];
+
+        // REPORTBACK IS CONDITIONAL
+        if (array_key_exists('posts', $signup)) {
+          $reportback_data = [
+              'id' => $signup['signup_id'],
+              'created_at' => strtotime($signup['created_at']),
+              'updated_at' => strtotime($signup['updated_at']),
+              'quantity' => $signup['quantity'],
+              'uri' => null,
+              'why_participated' => $signup['why_participated'],
+              'flagged' => null,
+          ];
+
+          // Add data for each reportback item
+          $reportback_item_array = [];
+          foreach ($signup['posts']['data'] as $post) {
+            $post_data = [
+              'id' => $post['id'],
+              'status' => dosomething_rogue_transform_rogue_status_to_phoenix_status($post['status']),
+              'caption' => $post['media']['caption'],
+              'uri' => null,
+              'media' => [
+                'uri' => $post['media']['url'],
+                'type' => 'image',
+              ],
+              'created_at' => strtotime($post['created_at']),
+              // @TODO: include kudo stuff here CONDITIONALLY
```

Comment:
@sbsmith86 ah another comment that needs to go! we actually _do_ want to include an empty array like this if there are no reactions, and we add any actual data in below